compiler-rt/lib/sanitizer_common/sanitizer_thread_arg_retval.cpp

```diff
@@ -53,7 +53,8 @@ void ThreadArgRetval::Finish(uptr thread, void* retval) {
 u32 ThreadArgRetval::BeforeJoin(uptr thread) const {
   __sanitizer::Lock lock(&mtx_);
   auto t = data_.find(thread);
-  CHECK(t);
+  if (!t)
+    return 0;
   CHECK(!t->second.detached);
```

Comment:
we still have the case when joined thread is reused by another 'detached' thread and we will hit the CHECK below

Reply:
A yes of course, will fix thanks!